## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 453525`
Commit: [`2a4ab2708f5ef15f5b61b74e4ebd965b86ed3bec`](https://github.com/NixOS/nixpkgs/commit/2a4ab2708f5ef15f5b61b74e4ebd965b86ed3bec) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/2a4ab2708f5ef15f5b61b74e4ebd965b86ed3bec..pull/453525/head))
Merge: [`56cf140d18d0b09cf5020f0e83fdaf0ff408897c`](https://github.com/NixOS/nixpkgs/commit/56cf140d18d0b09cf5020f0e83fdaf0ff408897c)

Logs: https://github.com/iedame/nixpkgs-review-gha/actions/runs/18630575456


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>rtabmap</li>
    <li>zed-open-capture</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>rtabmap</li>
    <li>zed-open-capture</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>zed-open-capture</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>zed-open-capture</summary>
<pre>cmake: enabled parallel building
cmake: enabled parallel installing
configurePhase completed in 40 seconds
Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
build flags: -j3 SHELL=/nix/store/j3wkpf582ngf1brv85jbgi1fk6c9lspq-bash-5.3p3/bin/bash
[ 66%] Building CXX object CMakeFiles/zed_open_capture.dir/src/videocapture.cpp.o
[ 66%] Building CXX object CMakeFiles/zed_open_capture.dir/src/sensorcapture.cpp.o
clang++: warning: argument unused during compilation: &#x27;-no-pie&#x27; [-Wunused-command-line-argument]
clang++: clang++warning: : argument unused during compilation: &#x27;-s&#x27; [-Wunused-command-line-argument]
warning: argument unused during compilation: &#x27;-no-pie&#x27; [-Wunused-command-line-argument]
clang++: warning: argument unused during compilation: &#x27;-s&#x27; [-Wunused-command-line-argument]
/nix/build/nix-10274-675047057/source/src/videocapture.cpp:32:10: fatal error: &#x27;linux/usb/video.h&#x27; file not found
   32 | #include &lt;linux/usb/video.h&gt;  // for UVC_GET_CUR, UVC_SET_CUR, UVC_GET_LEN
      |          ^~~~~~~~~~~~~~~~~~~
1 error generated.
make[2]: *** [CMakeFiles/zed_open_capture.dir/build.make:93: CMakeFiles/zed_open_capture.dir/src/videocapture.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:87: CMakeFiles/zed_open_capture.dir/all] Error 2
make: *** [Makefile:136: all] Error 2</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>zed-open-capture</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>zed-open-capture</summary>
<pre>-- Build files have been written to: /nix/build/nix-7655-2000738755/source/build
cmake: enabled parallel building
cmake: enabled parallel installing
Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
build flags: -j3 SHELL=/nix/store/xhsnsmvh1ka5mxszd1psxq36vaanb8jy-bash-5.3p3/bin/bash
[ 33%] Building CXX object CMakeFiles/zed_open_capture.dir/src/videocapture.cpp.o
[ 66%] Building CXX object CMakeFiles/zed_open_capture.dir/src/sensorcapture.cpp.o
clang++: warning: argument unused during compilation: &#x27;-no-pie&#x27; [-Wunused-command-line-argument]
clang++: warning: argument unused during compilation: &#x27;-s&#x27; [-Wunused-command-line-argument]
clang++: warning: argument unused during compilation: &#x27;-no-pie&#x27; [-Wunused-command-line-argument]
clang++: warning: argument unused during compilation: &#x27;-s&#x27; [-Wunused-command-line-argument]
/nix/build/nix-7655-2000738755/source/src/videocapture.cpp:32:10: fatal error: &#x27;linux/usb/video.h&#x27; file not found
   32 | #include &lt;linux/usb/video.h&gt;  // for UVC_GET_CUR, UVC_SET_CUR, UVC_GET_LEN
      |          ^~~~~~~~~~~~~~~~~~~
1 error generated.
make[2]: *** [CMakeFiles/zed_open_capture.dir/build.make:93: CMakeFiles/zed_open_capture.dir/src/videocapture.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:87: CMakeFiles/zed_open_capture.dir/all] Error 2
make: *** [Makefile:136: all] Error 2</pre>
</details>
</details>

--- a/pkgs/by-name/ze/zed-open-capture/package.nix
+++ b/pkgs/by-name/ze/zed-open-capture/package.nix
@@ -38,10 +38,16 @@ stdenv.mkDerivation {
     "-DBUILD_EXAMPLES=OFF"
   ];
 
-  meta = with lib; {
+  postPatch = ''
+    substituteInPlace CMakeLists.txt \
+      --replace-fail "cmake_minimum_required(VERSION 3.1)" "cmake_minimum_required(VERSION 3.10)"
+  '';
+
+  meta = {
     description = "Platform-agnostic camera and sensor capture API for the ZED 2, ZED 2i, and ZED Mini stereo cameras";
     homepage = "https://github.com/stereolabs/zed-open-capture";
-    license = licenses.mit;
-    maintainers = with maintainers; [ marius851000 ];
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ marius851000 ];
+    platforms = lib.platforms.linux;
   };
 }

